src/main/kotlin/com/mdrobnak/lalrpop/intentions/AddExplicitTypeIntention.kt

```diff
@@ -0,0 +1,64 @@
+package com.mdrobnak.lalrpop.intentions
+
+import com.intellij.codeInsight.intention.IntentionAction
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiFile
+import com.mdrobnak.lalrpop.LpLanguage
+import com.mdrobnak.lalrpop.psi.LpAlternative
+import com.mdrobnak.lalrpop.psi.LpElementFactory
+import com.mdrobnak.lalrpop.psi.LpNonterminal
+import com.mdrobnak.lalrpop.psi.getContextAndResolveType
+
+class AddExplicitTypeIntention : IntentionAction {
+    override fun startInWriteAction(): Boolean = true
+
+    override fun getText(): String = "Add explicit type"
+
+    override fun getFamilyName(): String = "Add explicit type"
+
+    override fun isAvailable(project: Project, editor: Editor?, file: PsiFile?): Boolean {
+        if (editor == null || file == null || file.language != LpLanguage) return false
+
+        var element = file.findElementAt(editor.caretModel.primaryCaret.offset)
+        while (element != null) {
+            if (element is LpNonterminal) return true
+            element = element.parent
+        }
+        return false
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?) {
+        if (editor == null || file == null || file.language != LpLanguage) return
+
+        var element = file.findElementAt(editor.caretModel.primaryCaret.offset)
+
+        var nonterminal: LpNonterminal? = null
+        var alternativeToUse: LpAlternative? = null
+
+        while (element != null) {
+            if (element is LpAlternative) {
```

Comment:
This changes the intention's behavior based on the element you select in the nonterminal. I think it should instead find the nonterminal element and then look at the first alternative to resolve the type. This would make the resolve behavior the same as in `WrongInferredTypeInspection`.

Reply:
> This changes the intention's behavior based on the element you select in the nonterminal.

This was by design.

Reply:
I'm concerned about the design being unintuitive. There is only one valid type for the nonterminal, so the intention should try to find this type. LALRPOP (and our `WrongInferredTypeInspection`) checks the type by going through the alternatives in order. Changing this behavior to being based on where the cursor doesn't seem to fit this model. Also, I don't think there are many other intentions which base their behavior on where the cursor is like this (available in all of the nonterminal's text, but silently changes behavior based on alternative).